xarray/indexes/range_index.py

```diff
@@ -84,9 +84,9 @@ def slice(self, sl: slice) -> "RangeCoordinateTransform":
         # TODO: support reverse transform (i.e., start > stop)?
         assert sl.start < sl.stop
 
-        new_size = (sl.stop - sl.start) // sl.step
```

Comment:
One trick I have learned is to use `range` objects to outsource this logic to CPython.

https://github.com/pydata/xarray/pull/7586

So you could do:
```python
newrange = range(self.size)[sl]
start = newrange.start
step = newrange.step or 1
size = len(newrange)
```

Reply:
Nice trick, this also allows supporting reverse indexing without additional effort.

Reply:
Actually, I remember the real API is `slice.indices(size)` so an alternative is :
```
        start, stop, step = slicer.indices(self.size)
        size = len(range(start, stop, step))
```

I think you still need to construct the range object to get the right `size` so not much difference